pkg/oc/lib/groupsync/rfc2307/ldapinterface.go

```diff
@@ -142,7 +143,12 @@ func (e *LDAPInterface) ListGroups() ([]string, error) {
 	for _, group := range groups {
 		ldapGroupUID := ldaputil.GetAttributeValue(group, []string{e.groupQuery.QueryAttribute})
 		if len(ldapGroupUID) == 0 {
-			return nil, fmt.Errorf("unable to find LDAP group UID for %s", group)
```

Comment:
If group is of `ldap.v2.Entry` type then just change `%s` to `%#v`, I don't think `json.Marshal` is needed here.

Reply:
@soltysh that's what I tried originally, the test there fails because of the "Attributes" thing being a pointer rather than a direct reference:

```
--- FAIL: TestListGroups (0.00s)
    ldapinterface_test.go:288: no UID on entry: incorrect error returned:
        	expected:
        	unable to find LDAP group UID for &{cn=,ou=groups,dc=example,dc=com [0xc000132440]}
        	got:
        	unable to find LDAP group UID for &{cn=,ou=groups,dc=example,dc=com [0xc000132400]}
=== RUN   TestUserEntryFor
```
Is there something I'm missing? I'm fairly new to Golang itself, sorry.

Reply:
`return nil, fmt.Errorf("unable to find LDAP group UID for %s", group.DN)`

Reply:
I see, that's much better. Thank you very much for the pointer @enj, didn't know I could use this.

Reply:
I fixed it, please look over it again when you have time, thanks!